### PR TITLE
Improve recovery of misplaced attributes on statements

### PR DESCRIPTION
**Pull request by ahoppen**
Fixes apple/swift-syntax#714
rdar://99594971

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Tests/SwiftParserTest/translated/SwitchTests.swift [hunk 1]
```diff
@@ -950,7 +950,7 @@ final class SwitchTests: XCTestCase {
       }
       """,
       diagnostics: [
-        DiagnosticSpec(message: "expected label in switch case")
+        DiagnosticSpec(message: "unexpected code '@moreGarbage' in switch case")
```
Comment (bnbarham) [inline Tests/SwiftParserTest/translated/SwitchTests.swift:953]: Do we parse any single attribute rather than just `@unknown`?

**Review by bnbarham — APPROVED** [review]

File: Tests/SwiftParserTest/translated/SwitchTests.swift (hunk 1 above)
Comment (ahoppen) [reply Tests/SwiftParserTest/translated/SwitchTests.swift:953]: Yes, we do. That’s what the syntax tree currently allows.

**Comment by ahoppen** [comment]
@swift-ci Please test Linux

**Pull request merged**